akka-remote/src/main/scala/akka/remote/Remoting.scala

```diff
@@ -533,25 +533,30 @@ private[remote] class EndpointManager(conf: Config, log: LoggingAdapter) extends
       }
       Future.fold(allStatuses)(true)(_ && _) map ManagementCommandAck pipeTo sender()
 
-    case Quarantine(address, uidOption) ⇒
+    case Quarantine(address, uidToQuarantineOption) ⇒
       // Stop writers
-      endpoints.writableEndpointWithPolicyFor(address) match {
-        case Some(Pass(endpoint, _, _)) ⇒
+      (endpoints.writableEndpointWithPolicyFor(address), uidToQuarantineOption) match {
+        case (Some(Pass(endpoint, _, _)), None) ⇒
           context.stop(endpoint)
-          if (uidOption.isEmpty) {
-            log.warning("Association to [{}] with unknown UID is reported as quarantined, but " +
-              "address cannot be quarantined without knowing the UID, gating instead for {} ms.",
-              address, settings.RetryGateClosedFor.toMillis)
-            endpoints.markAsFailed(endpoint, Deadline.now + settings.RetryGateClosedFor)
-          }
-        case _ ⇒ // nothing to stop
+          log.warning("Association to [{}] with unknown UID is reported as quarantined, but " +
+            "address cannot be quarantined without knowing the UID, gating instead for {} ms.",
+            address, settings.RetryGateClosedFor.toMillis)
+          endpoints.markAsFailed(endpoint, Deadline.now + settings.RetryGateClosedFor)
+        case (Some(Pass(endpoint, Some(currentUid), _)), Some(quarantineUid)) if currentUid == quarantineUid ⇒
```

Comment:
(plus, backtick is not really available on my keyboard so I tend to avoid it :) )